test/box/gh-5998-one-tx-for-ddl-errinj.test.lua

```diff
@@ -24,23 +27,20 @@ function create_sk_in_tx(space, name, channel)
     box.begin()
     errinj.set('ERRINJ_BUILD_INDEX_DELAY', true)
     space:create_index(name)
-    channel:get()
     box.commit()
+    assert(channel:is_full() == false)
     channel:put(true)
 end
 _ = test_run:cmd("setopt delimiter ''");
 
-f1 = fiber.new(create_sk_in_tx, box.space.s1, "sk1", channel1)
-f2 = fiber.new(create_sk_in_tx, box.space.s2, "sk2", channel2)
-fiber.sleep(0)
+f1 = fiber.create(create_sk_in_tx, box.space.s1, "sk1", channel1)
+f2 = fiber.create(create_sk_in_tx, box.space.s2, "sk2", channel2)
 
 errinj.set('ERRINJ_BUILD_INDEX_DELAY', false);
-channel1:put(true)
-fiber.sleep(0.01)
-channel1:get()
+while channel1:get(0.001) == nil do fiber.sleep(0) end
```

Comment:
Fixed. Also applied this diff:
```
diff --git a/test/box/gh-5998-one-tx-for-ddl-errinj.test.lua b/test/box/gh-5998-one-tx-for-ddl-errinj.test.lua
index 9d565e306..9b00d9731 100644
--- a/test/box/gh-5998-one-tx-for-ddl-errinj.test.lua
+++ b/test/box/gh-5998-one-tx-for-ddl-errinj.test.lua
@@ -27,7 +27,7 @@ function create_sk_in_tx(space, name, channel)
     box.begin()
     errinj.set('ERRINJ_BUILD_INDEX_DELAY', true)
     space:create_index(name)
-    box.commit()
+    _ = pcall(box.commit)
     assert(channel:is_full() == false)
     channel:put(true)
 end
@@ -37,7 +37,8 @@ f1 = fiber.create(create_sk_in_tx, box.space.s1, "sk1", channel1)
 f2 = fiber.create(create_sk_in_tx, box.space.s2, "sk2", channel2)
 
 errinj.set('ERRINJ_BUILD_INDEX_DELAY', false);
-while channel1:get(0.001) == nil do fiber.sleep(0) end
+channel1:get()
+channel2:get()
```
`channel2:get()` is required to make sure that creation of secondary index is finished (of the second space). Otherwise extremely rarely we may get:
```
001] +++ var/rejects/box/gh-5998-one-tx-for-ddl-errinj.reject	Wed Sep  1 16:50:50 2021
[001] @@ -101,6 +101,7 @@
[001]   | ...
[001]  box.space.s2:drop()
[001]   | ---
[001] + | - error: 'Can''t modify space ''s2'': the space is already being modified'
[001]   |
```